plugins/webp-uploads/hooks.php

```diff
@@ -538,7 +549,27 @@ function webp_uploads_update_image_references( $content ) {
  * @return string The updated img tag.
  */
 function webp_uploads_img_tag_update_mime_type( $original_image, $context, $attachment_id ) {
-	$image    = $original_image;
+	$image = $original_image;
+
+	/**
+	 * Metadata potentially amended by webp_uploads_create_sources_property().
+	 *
+	 * Note the sources key is not normally present in the response for wp_get_attachment_metadata(). The sources
+	 * key here, however, is being injected via the 'wp_generate_attachment_metadata' filter via the
+	 * webp_uploads_create_sources_property() function.
+	 *
+	 * @see webp_uploads_create_sources_property()
+	 *
+	 * @var array{
+	 *          width: int,
+	 *          height: int,
+	 *          file: non-falsy-string,
+	 *          sizes: array,
+	 *          image_meta: array,
+	 *          filesize: int,
+	 *          sources?: array<string, array{ file: string, filesize: int }>
+	 *      } $metadata
+	 */
```

Comment:
Without this type definition, the `isset( $metadata['sources'][ $target_mime ]['file'] )` check below fails PHPStan because `wp_get_attachment_metadata()` doesn't normally return a `sources` key.

Reply:
I revisited this when bumping PHPStan to level 3. See #1200.